plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java

```diff
@@ -0,0 +1,458 @@
+        PlanMatchPattern topNOverTableScan = node(TopNNode.class, anyTree(node(TableScanNode.class)));
+
+        try (TestTable testTable = new TestTable(
+                getQueryRunner()::execute,
+                "test_case_sensitive_topn_pushdown",
+                "(a_string varchar(10), a_char char(10), a_bigint bigint) with ( primary_key = ARRAY['a_bigint'])",
+                List.of(
+                        "'A', 'A', 1",
+                        "'B', 'B', 2",
+                        "'a', 'a', 3",
+                        "'b', 'b', 4"))) {
+            assertConditionallyOrderedPushedDown(
+                    getSession(),
+                    "SELECT a_bigint FROM " + testTable.getName() + " ORDER BY a_string ASC LIMIT 2",
+                    expectTopNPushdown,
+                    topNOverTableScan);
+            assertConditionallyOrderedPushedDown(
+                    getSession(),
+                    "SELECT a_bigint FROM " + testTable.getName() + " ORDER BY a_string DESC LIMIT 2",
+                    expectTopNPushdown,
+                    topNOverTableScan);
+            assertConditionallyOrderedPushedDown(
+                    getSession(),
+                    "SELECT a_bigint FROM " + testTable.getName() + " ORDER BY a_char ASC LIMIT 2",
+                    expectTopNPushdown,
+                    topNOverTableScan);
+            assertConditionallyOrderedPushedDown(
+                    getSession(),
+                    "SELECT a_bigint FROM " + testTable.getName() + " ORDER BY a_char DESC LIMIT 2",
+                    expectTopNPushdown,
+                    topNOverTableScan);
+
+            // multiple sort columns with at-least one case-sensitive column
+            assertConditionallyOrderedPushedDown(
+                    getSession(),
+                    "SELECT a_bigint FROM " + testTable.getName() + " ORDER BY a_bigint, a_char LIMIT 2",
+                    expectTopNPushdown,
+                    topNOverTableScan);
+            assertConditionallyOrderedPushedDown(
+                    getSession(),
+                    "SELECT a_bigint FROM " + testTable.getName() + " ORDER BY a_bigint, a_string DESC LIMIT 2",
+                    expectTopNPushdown,
+                    topNOverTableScan);
+        }
+    }
+
+    private void assertConditionallyOrderedPushedDown(
+            Session session,
+            @Language("SQL") String query,
+            boolean condition,
+            PlanMatchPattern otherwiseExpected)
+    {
+        QueryAssertions.QueryAssert queryAssert = assertThat(query(session, query)).ordered();
+        if (condition) {
+            queryAssert.isFullyPushedDown();
+        }
+        else {
+            queryAssert.isNotFullyPushedDown(otherwiseExpected);
+        }
+    }
+
+    @Override
+    protected SqlExecutor onRemoteDatabase()
+    {
+        return igniteServer::execute;
+    }
+
+    @Override
+    public void testNullSensitiveTopNPushdown()
```

Comment:
Is it overridden for the primary key specification? Maybe we can modify the base test class to allow using a provided impl of TestTable which will add table properties.